### ISSUE TITLE
Build Warning on Linux Release

### ISSUE THREAD

**Issue by Myoldmopar** [NREL/EnergyPlus#7314]
Issue overview
--------------
```
/eplus/repos/jmarrec/src/EnergyPlus/WeatherManager.cc: In function ‘void EnergyPlus::WeatherManager::SetCurrentWeather()’:
/eplus/repos/jmarrec/src/EnergyPlus/WeatherManager.cc:2490:34: warning: ‘%02d’ directive writing between 2 and 11 bytes into a region of size between 0 and 4 [-Wformat-overflow=]
         std::sprintf(time_stamp, "%02d/%02d %02d:", Month, DayOfMonth, HourOfDay - 1);
                                  ^~~~~~~~~~~~~~~~~
/eplus/repos/jmarrec/src/EnergyPlus/WeatherManager.cc:2490:34: note: directive argument in the range [-2147483648, 2147483646]
In file included from /usr/include/stdio.h:873,
                 from /usr/include/c++/8/cstdio:42,
                 from /usr/include/c++/8/ext/string_conversions.h:43,
                 from /usr/include/c++/8/bits/basic_string.h:6391,
                 from /usr/include/c++/8/string:52,
                 from /usr/include/c++/8/stdexcept:39,
                 from /usr/include/c++/8/array:39,
                 from /eplus/repos/jmarrec/src/EnergyPlus/WeatherManager.cc:49:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:36:34: note: ‘__builtin___sprintf_chk’ output between 10 and 37 bytes into a destination of size 10
   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       __bos (__s), __fmt, __va_arg_pack ());
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### Details
Some additional details for this issue (if relevant):
 - Platform: Ubuntu 18.10
 - GCC: 8.2.0

**Comment by Myoldmopar** [comment]
@jasondegraw I believe this would be easy for you to fix, as you've been in the weather manager most recently.  If not, let me know and I'll take a look or reassign.  It could be valid code and the warning should just be muted, and I'm OK with that if it is justified.

**Comment by tidecoder** [comment]
@Myoldmopar What config options did you apply to build EnergyPlus? 
I have Ubuntu Desktop 18.10 and GCC version 8.3.0. 
So far my build and compile are good without warning with following configure options:

`
BUILD_DOCS    OFF
BUILD_FORTRAN    OFF
BUILD_GMOCK   OFF
BUILD_GTEST    ON
BUILD_PACKAGE    OFF
BUILD_PERFORMACE_TESTS    OFF
BUILD_TESTING    ON
BUILD_WCE_TESTING    OFF
CMAKE_BUILD_TYPE    Debug
ENABLE_COVERAGE    OFF
`
I am thinking are above config options enough?

**Comment by Myoldmopar** [comment]
This is a release build on Linux, not a debug build.  Set CMAKE_BUILD_TYPE to Release and you should see this.

**Comment by YanfeiNREL** [comment]
The issue is: "Month, DayOfMonth, HourOfDay" are all integers; while the "%02d/%02d %02d" formatting is trying to compress those integers into 2 digitals. However, the compiler is not smart enough to recognize this behavior. Basically i cast them into short integer. Please see the branch @"WarningLinuxRelease7314"

**Comment by Myoldmopar** [comment]
@YanfeiNREL thanks for the fix, I'll be checking out the branch.  I'm going to reopen this issue until the branch is merged into the main code base.

**Comment by Myoldmopar** [comment]
Fixed via #7373
